src/OpenTelemetry.Exporter.Prometheus.Shared/PrometheusExporter.cs

```diff
@@ -69,23 +60,22 @@ public PrometheusExporter(PrometheusExporterOptions options)
             set => this.funcExport = value;
         }
 
+        internal Action OnDispose { get; set; }
+
         internal PrometheusCollectionManager CollectionManager { get; }
 
+        /// <inheritdoc/>
         public override ExportResult Export(in Batch<Metric> metrics)
         {
             return this.OnExport(metrics);
         }
 
+        /// <inheritdoc/>
         protected override void Dispose(bool disposing)
         {
-            if (!this.disposed)
+            if (disposing)
```

Comment:
I'm sorry I might have missed this so let me know if this has been discussed before. Why are we getting rid of check on `disposed` here?

Reply:
Not sure! @Yun-Ting Was this intentional? I might have done it, can't remember 😄 But probably safe to put back?

Reply:
Adding the private bool back as `Dispose` could be called multiple times.